pkg/aggregator/aggregator_test.go

```diff
@@ -142,3 +146,113 @@ func TestSetHostname(t *testing.T) {
 	assert.Equal(t, "different-hostname", agg.hostname)
 	assert.Equal(t, "different-hostname", checkSender.defaultHostname)
 }
+
+func TestDefaultData(t *testing.T) {
+	resetAggregator()
+	s := &serializer.MockSerializer{}
+	agg := InitAggregator(s, "hostname", "agent")
+	start := time.Now()
+
+	s.On("SendServiceChecks", metrics.ServiceChecks{{
+		CheckName: "datadog.agent.up",
+		Status:    metrics.ServiceCheckOK,
+		Ts:        start.Unix(),
+		Host:      agg.hostname,
+	}}).Return(nil).Times(1)
+
+	series := metrics.Series{&metrics.Serie{
+		Name:           fmt.Sprintf("datadog.%s.running", agg.agentName),
+		Points:         []metrics.Point{{Value: 1, Ts: float64(start.Unix())}},
+		Tags:           []string{fmt.Sprintf("version:%s", version.AgentVersion)},
+		Host:           agg.hostname,
+		MType:          metrics.APIGaugeType,
+		SourceTypeName: "System",
+	}, &metrics.Serie{
+		Name:           fmt.Sprintf("n_o_i_n_d_e_x.datadog.%s.payload.dropped", agg.agentName),
+		Points:         []metrics.Point{{Value: 0, Ts: float64(start.Unix())}},
+		Host:           agg.hostname,
+		MType:          metrics.APIGaugeType,
+		SourceTypeName: "System",
+	}}
+
+	s.On("SendSeries", series).Return(nil).Times(1)
+
+	agg.flush(start)
+	s.AssertNotCalled(t, "SendEvents")
+	s.AssertNotCalled(t, "SendSketch")
+}
+
+func TestRecurentSeries(t *testing.T) {
+	resetAggregator()
```

Comment:
I think `resetAggregator` should also re-initialize `recurrentSeries` to reset the aggregator package to a clean state.

Again, this feels a bit like a code smell, but this'll do for now :)